lib/matplotlib/tests/test_axes.py

```diff
@@ -145,6 +145,24 @@ def test_twinx_cla():
     assert ax.yaxis.get_visible()
 
 
+@cleanup
+def test_twin_inherit_autoscale():
```

Comment:
As much as I dislike image tests, do you have one of those, or some other inspection of the axis to check that the ticks are acting the way you expect?